From 54a0dbb8a8b0819ac77755fe614131c5ba60401b Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:09:16 -0400
Subject: [usb] ftdi_sio: add NovaTech OrionLXm product ID

Message-id: <1410887463-20674-485-git-send-email-dzickus@redhat.com>
Patchwork-id: 95545
O-Subject: [RHEL7 PATCH 484/591] USB: ftdi_sio: add NovaTech OrionLXm product ID
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit d0839d757e6294921c31b1c4ca4f1dcc5df63bcd
Author: George McCollister <george.mccollister@gmail.com>
Date:   Sat Apr 26 12:04:47 2014 +0200

    USB: ftdi_sio: add NovaTech OrionLXm product ID

    The NovaTech OrionLXm uses an onboard FTDI serial converter for JTAG and
    console access.

    Here is the lsusb output:
    Bus 004 Device 123: ID 0403:7c90 Future Technology Devices
    International, Ltd

    Signed-off-by: George McCollister <george.mccollister@gmail.com>
    Cc: stable <stable@vger.kernel.org>
    Signed-off-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/ftdi_sio.c b/drivers/usb/serial/ftdi_sio.c
index 9333c98..d69c5a5 100644
--- a/drivers/usb/serial/ftdi_sio.c
+++ b/drivers/usb/serial/ftdi_sio.c
@@ -577,6 +577,8 @@ static const struct usb_device_id id_table_combined[] = {
  { USB_DEVICE(FTDI_VID, FTDI_TAVIR_STK500_PID) },
  { USB_DEVICE(FTDI_VID, FTDI_TIAO_UMPA_PID),
   .driver_info = (kernel_ulong_t)&ftdi_jtag_quirk },
+ { USB_DEVICE(FTDI_VID, FTDI_NT_ORIONLXM_PID),
+  .driver_info = (kernel_ulong_t)&ftdi_jtag_quirk },
  /*
   * ELV devices:
   */
diff --git a/drivers/usb/serial/ftdi_sio_ids.h b/drivers/usb/serial/ftdi_sio_ids.h
index 0e7e5f2..a387244 100644
--- a/drivers/usb/serial/ftdi_sio_ids.h
+++ b/drivers/usb/serial/ftdi_sio_ids.h
@@ -531,6 +531,11 @@
  */
 #define FTDI_TIAO_UMPA_PID 0x8a98 /* TIAO/DIYGADGET USB Multi-Protocol Adapter */
 
+/*
+ * NovaTech product ids (FTDI_VID)
+ */
+#define FTDI_NT_ORIONLXM_PID 0x7c90 /* OrionLXm Substation Automation Platform */
+
 
 /********************************/
 /** third-party VID/PID combos **/
-- 
1.7.1